crates/matrix-sdk-base/src/sync.rs

```diff
@@ -90,13 +90,22 @@ impl RoomUpdates {
     /// Iterate over all room IDs, from [`RoomUpdates::left`],
     /// [`RoomUpdates::joined`], [`RoomUpdates::invited`] and
     /// [`RoomUpdates::knocked`].
-    pub(crate) fn iter_all_room_ids(&self) -> impl Iterator<Item = &OwnedRoomId> {
+    pub fn iter_all_room_ids(&self) -> impl Iterator<Item = &OwnedRoomId> {
         self.left
             .keys()
             .chain(self.joined.keys())
             .chain(self.invited.keys())
             .chain(self.knocked.keys())
     }
+
+    /// Returns whether or not this update contains any changes to the list
+    /// of invited, joined, knocked or left rooms.
+    pub fn is_empty(&self) -> bool {
+        self.invited.is_empty()
+            && self.joined.is_empty()
+            && self.knocked.is_empty()
+            && self.left.is_empty()
+    }
```

Comment:
We can put this into a separate PR, those methods are useful on their own.

Don't forget a changelog entry.